include/mbedtls/mbedtls_config.h

```diff
@@ -1798,8 +1798,9 @@
  * Requires: MBEDTLS_PSA_CRYPTO_C
  *
  * \note TLS 1.3 uses PSA crypto for cryptographic operations that are
- *       directly performed by TLS 1.3 code. As a consequence, you must
- *       call psa_crypto_init() before the first TLS 1.3 handshake.
+ *       directly performed by TLS 1.3 code. As a consequence, when TLS 1.3
+ *       is enabled, a TLS handshake may call psa_crypto_init(), even
```

Comment:
Should we have a changelog entry announcing that `psa_crypto_init` will now sometimes be called when starting a TLS connection (even a 1.2 connection)?

Reply:
What's observable from a user perspective about `psa_crypto_init()` being called?

Reply:
> what's observable from a user perspective about psa_crypto_init() being called?

* It can fail. Unlikely in an application that's doing TLS, but possible.
    * Entropy failure (but how are you getting entropy for the RNG passed to TLS).
    * Stray file that screws up PSA storage initialization (but how would that stray file have been created).
    * Low resources (but you don't really stand a chance of doing a TLS connection then).
* It may leave resources around at the end, so a memory leak detector would complain. (That's why we're calling `mbedtls_psa_crypto_free` unconditionally in the tests.)

Those are pretty exotic concerns in an application that's doing a TLS connection, which is why I'm not sure.

Reply:
Then I think as a precaution we should add a brief ChangeLog entry about it.